phpBB/install/startup.php

```diff
@@ -47,6 +47,48 @@ function phpbb_include_updated($path, $phpbb_root_path, $optional = false)
 	}
 }
 
+function installer_msg_handler($errno, $msg_text, $errfile, $errline)
+{
+	global $phpbb_installer_container;
+
+	switch ($errno)
+	{
+		case E_NOTICE:
+		case E_WARNING:
+		case E_USER_WARNING:
+		case E_USER_NOTICE:
+			$msg = '[phpBB debug] "' . $msg_text . '" in file ' . $errfile . ' on line ' . $errline;
+
+			try
+			{
+				/** @var \phpbb\install\helper\iohandler\iohandler_interface $iohandler */
+				$iohandler = $phpbb_installer_container->get('installer.helper.iohandler');
+				$iohandler->add_warning_message($msg);
+			}
+			catch (\phpbb\install\helper\iohandler\exception\iohandler_not_implemented_exception $e)
+			{
+				print ($msg);
```

Comment:
it will try to use the iohandler, if it is not set, it is more then likely an HTML page with no AJAX so it should be fine.